implementation/src/main/java/io/smallrye/graphql/GraphQLExtension.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.graphql;
+
+import java.lang.reflect.Type;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessBean;
+
+import org.eclipse.microprofile.graphql.GraphQLApi;
+
+import graphql.schema.GraphQLSchema;
+import io.leangen.graphql.GraphQLSchemaGenerator;
+import io.leangen.graphql.metadata.strategy.query.AnnotatedResolverBuilder;
+
+/**
+ * This is a CDI extension that detects GraphQL components
+ * and generate the GraphQL Schema
+ * 
+ * @author Phillip Kruger (phillip.kruger@redhat.com)
+ */
+public class GraphQLExtension implements Extension {
```

Comment:
Is there a way to do this without needing a CDI Extension?

Thinking from the perspective of what's needed to have it running in Quarkus using ArC

Reply:
There might be. I just need to find all classes with a certain annotation on startup. I can look into that at some point.

Reply:
I looked at Jandex. I see that is used in OpenApi. I am going to need more time to see if that can work. And I might need to post on the forum for help. Can we merge like this, so then we can continue with the MP TCK ? Then I create a issue in GitHub to look at Jandex or another alternative. Le me know. Thanks

Reply:
@kenfinnigan - any feedback ?